xds/internal/balancer/cdsbalancer/cluster_handler.go

```diff
@@ -261,59 +309,55 @@ func (c *clusterNode) handleResp(clusterUpdate xdsresource.ClusterUpdate, err er
 	// the update to build (ex. if a child is created and a watch is started,
 	// that child hasn't received an update yet due to the mutex lock on this
 	// callback).
-	var createdChild, deletedChild bool
+	var createdChild bool
 
 	// This map will represent the current children of the cluster. It will be
 	// first added to in order to represent the new children. It will then have
-	// any children deleted that are no longer present. Then, from the cluster
-	// update received, will be used to construct the new child list.
-	mapCurrentChildren := make(map[string]*clusterNode)
+	// any children deleted that are no longer present.
+	mapCurrentChildren := make(map[string]bool)
 	for _, child := range c.children {
-		mapCurrentChildren[child.clusterUpdate.ClusterName] = child
+		mapCurrentChildren[child] = true
 	}
 
 	// Add and construct any new child nodes.
 	for child := range newChildren {
 		if _, inChildrenAlready := mapCurrentChildren[child]; !inChildrenAlready {
-			createdChild = true
-			mapCurrentChildren[child] = createClusterNode(child, c.clusterHandler.parent.xdsClient, c.clusterHandler)
+			if c.depth == maxDepth-1 {
```

Comment:
Actually, depth is not related to whether the children list is different, right? If this already exceeds the maxdepth, and receives another aggregated cluster update later, it's still the same error. Should we not send the second error?

Reply:
Ah what a catch! Moved to the loop in for _, childName := range clusterUpdate.PrioritizedClusterNames. Putting it inside makes it so empty child list doesn't trigger it. Yeah, I agree that another resp for the cluster that causes it to exceed depth should cause an error to be written to the buffer. However, I still think that another resp for another cluster as it builds cluster update shouldn't cause error to be written to buffer.

Reply:
Also added assertion in test covering this.